modules/post/multi/gather/aws_ec2_instance_metadata.rb

```diff
@@ -36,7 +36,7 @@ def initialize(info = {})
 
   def check_aws_metadata
     resp = simple_get(@target_uri)
-    unless resp =~ /^instance-id.$/m
+    unless resp =~ /^instance-id$/m
```

Comment:
In the original the purpose of the `.` was to match the newline after `instance-id` (this is what the `m` regex modifier means), although in my testing it looks like it works just fine without it. Since we don't have any wildcard matches now, we can remove the `m` as well.

Reply:
Perhaps this was a lazy hack to check for rogue `\r` before line end `\n`; rather than using `resp.to_s.strip` ?